src/Form/MetadataDisplayForm.php

```diff
@@ -389,17 +412,12 @@ public static function ajaxPreview($form, FormStateInterface $form_state) {
             '#type' => 'details',
             '#open' => TRUE,
             '#title' => 'HTML Output',
-            'messages' => [
-              '#markup' => $message,
-              '#attributes' => [
-                'class' => ['error'],
-              ],
-            ],
             'render' => [
                 '#markup' => $render,
             ],
           ];
         }
+	$output = static::buildAjaxPreviewError($output, $message);
```

Comment:
@aksm sorry for being picky here, but it makes more sense to to leave the logic of "is there a message to be shown" to the caller than to the actual buildAjaxPreviewError method. means the  if(!empty($message)) { $output = static::buildAjaxPreviewError($output, $message);
}

Also, not sure if passing the whole rendered (big variable) around to another method to add a key and then back (these are not Objects so you are actually passing a huge value/memory) is the best approach. 

What if static::buildAjaxPreviewError just generates the piece. And then you add to the existing $output that key?

Does this make sense to you?

Reply:
All makes sense and good to know. Will make the changes and ping you. Thanks!